tls/s2n_client_hello.c

```diff
@@ -111,6 +111,34 @@ ssize_t s2n_client_hello_get_cipher_suites_length(struct s2n_client_hello *ch) {
     return ch->cipher_suites.size;
 }
 
+int s2n_client_hello_cb_done(struct s2n_connection *conn)
+{
+    struct s2n_client_hello *client_hello;
+
+    POSIX_ENSURE_REF(conn);
+    POSIX_ENSURE_REF(conn->config);
+    S2N_ERROR_IF(conn->config->client_hello_cb_mode !=
+        S2N_CLIENT_HELLO_CB_NONBLOCKING, S2N_ERR_INVALID_STATE);
+    GUARD_NONNULL(client_hello = s2n_connection_get_client_hello(conn));
+    S2N_ERROR_IF(client_hello->parsed == 0, S2N_ERR_ASYNC_NOT_PERFORMED);
```

Comment:
I missed these all over your code the first time. We've changed how we handle GUARDs recently:
* We prefer the GUARD/ENSURE macros prefixed with the return type context. So POSIX_ENSURE_REF instead of GUARD_NONNULL. You can see the new macros here: https://github.com/aws/s2n-tls/blob/main/utils/s2n_safety_macros.h We're kind of halfway through a migration, so you can use this script to change your code automatically: There's a codemod script you can also run against your code, to automate the latest partial migration: https://github.com/aws/s2n-tls/blob/main/scripts/s2n_safety_explicit_context/codemod.sh
* We don't use S2N_ERROR_IF anymore. We prefer POSIX_ENSURE, since you don't have to invert your expected condition for it.

Reply:
sure. Ran the script, applied it's suggestions. replaced `S2N_ERROR_IF` to `POSIX_ENSURE`